Julia does not build when /bin/sh is not bash

When /bin/sh is dash rather than bash, `make` fails with
```
make[2]: Leaving directory '/var/tmp/portage/dev-lang/julia-9999/work/julia-9999/src/flisp'
 ./flisp/flisp ./mk_julia_flisp_boot.scm
type error: file: expected string, got #f
in file ./mk_julia_flisp_boot.scm
Makefile:90: recipe for target 'julia_flisp.boot' failed
make[1]: *** [julia_flisp.boot] Error 1
```
because `export julia_flisp.boot=$(BUILDDIR)/julia_flisp.boot` in `src/Makefile` does not work in other shells due to the dot in the variable name. One way to solve the issue is by setting `SHELL = /bin/bash` in the Makefile. Another is by removing the dot from the Makefile and `mk_julia_flisp_boot.scm`.

--- a/src/Makefile
+++ b/src/Makefile
@@ -84,8 +84,8 @@ $(BUILDDIR)/%.dbg.obj: %.cpp $(HEADERS) $(shell which $(LLVM_CONFIG)) | $(BUILDD
 $(BUILDDIR)/julia_flisp.boot.inc: $(BUILDDIR)/julia_flisp.boot $(FLISP_EXECUTABLE)
 	@$(call PRINT_FLISP, $(call spawn,$(FLISP_EXECUTABLE)) ./bin2hex.scm < $< > $@)
 
-export julia_flisp.boot=$(BUILDDIR)/julia_flisp.boot
-$(julia_flisp.boot): julia-parser.scm julia-syntax.scm \
+export julia_flisp_boot=$(BUILDDIR)/julia_flisp.boot
+$(julia_flisp_boot): julia-parser.scm julia-syntax.scm \
 	match.scm utils.scm jlfrontend.scm mk_julia_flisp_boot.scm $(FLISP_EXECUTABLE)
 	@$(call PRINT_FLISP, $(call spawn,$(FLISP_EXECUTABLE)) ./mk_julia_flisp_boot.scm)
 

--- a/src/mk_julia_flisp_boot.scm
+++ b/src/mk_julia_flisp_boot.scm
@@ -1,2 +1,2 @@
 (load "jlfrontend.scm")
-(make-system-image (os.getenv "julia_flisp.boot"))
+(make-system-image (os.getenv "julia_flisp_boot"))

